Fix 1275: Make MaxValueValidator default an integer

wearebasti opened the pull request:
This should fix the issue #1275 by making the default value an integer and providing the option to set the max value per filter instance

carltongibson reviewed (commented):
Hi. 

I deliberately don't want to add a kwarg. It complicates the interface for all users, only 1 in a million of which will ever need to change it. Create a subclass if that's you. (Such would need docs and tests.) 

Can you add a test case showing the problem case this change would resolve? (Thinking through trapping the signal there...) 

Cheers.

wearebasti commented:
Hi, thanks for the input!

> I deliberately don't want to add a kwarg. It complicates the interface for all users, only 1 in a million of which will ever need to change it. Create a subclass if that's you. (Such would need docs and tests.)

Removed the kwarg again, just thought it might come in handy for some people - no feelings in either way from my side.

> Can you add a test case showing the problem case this change would resolve? (Thinking through trapping the signal there...)
> 
I added a test-case that activates the Decimal Trap first. The idea is to avoid some - for the user - strange side-effects of float/Decimal/int-comparisons (https://docs.python.org/3/library/decimal.html#floating-point-notes).

Does this make sense for you?

carltongibson commented on django_filters/filters.py:
```diff
@@ -362,7 +362,7 @@ def get_max_validator(self):
         Return a MaxValueValidator for the field, or None to disable.
         """
-        return MaxValueValidator(1e50)
+        return MaxValueValidator(int(1e50))
```
I just want to have a look at the order of validation in the field, but maybe this should be a decimal already... 
(i.e. did we already run the value from the widget through `to_python`, giving us a decimal in the `value < limit` check...)

carltongibson reviewed (commented):
Super. Thanks for the update. Let me have a play, but looks good.
